### PR TITLE
shipit_code_coverage: Don't do anything if the coverage report has already been generated and uploaded

### LINKED ISSUE #1118: shipit_code_coverage: Make the task idempotent
Blocks #704 and #954.

This is also needed because of #1115.

Solving this is just a matter of checking whether the given revision is already present in codecov.io.

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #1118.

### CODE REVIEW

File: src/shipit_code_coverage/shipit_code_coverage/codecov.py [hunk 1]
```diff
@@ -169,6 +169,15 @@ def generate_chunk_mapping(self):
             tar.add('chunk_mapping.sqlite')
 
     def go(self):
+        if self.from_pulse:
+            commit_sha = self.githubUtils.get_commit(self.revision)
```
Comment (jankeromnes) [inline src/shipit_code_coverage/shipit_code_coverage/codecov.py]: Nit: Shouldn't you move the definition of `commit_sha` above the `if`, in order for it to still be defined in `self.githubUtils.post_github_status(commit_sha)` below?

File: src/shipit_code_coverage/tests/test_uploader.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
```
Comment (jankeromnes) [inline src/shipit_code_coverage/tests/test_uploader.py]: Nit: Maybe we should release all test files in this repository to the Public Domain, just like [all Mozilla test files](https://www.mozilla.org/en-US/MPL/license-policy/#Licensing_of_Mozilla_Code)?

To do that, you could add the following header comment:

    # Any copyright is dedicated to the Public Domain.
    # http://creativecommons.org/publicdomain/zero/1.0/

File: src/shipit_code_coverage/tests/test_uploader.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from shipit_code_coverage import uploader
+
+
+@responses.activate
+def test_get_codecov(mock_secrets, codecov_commits):
+    data = uploader.get_codecov('0548f006a32c138218ad586177db83486c82a09e')
+    assert data['commit']['state'] == 'complete'
+
+    with pytest.raises(requests.exceptions.HTTPError):
+        uploader.get_codecov('7d7b21f5ede24dbdd59e17980a332f4b240f8e13')
```
Comment (jankeromnes) [inline src/shipit_code_coverage/tests/test_uploader.py]: I'm not sure I understand this `with` statement properly: Does it assert that an `HTTPError` is actually thrown? Shouldn't you check that the return value of `uploader.get_codecov` is `None`?

Comment (marco-c) [reply src/shipit_code_coverage/tests/test_uploader.py]: Yes, it asserts that `HTTPError` is thrown (it's thrown by https://github.com/mozilla-releng/services/pull/1122/files#diff-0db15d7e275ece3900d2c3f084332f1fR75, as when a commit is missing codecov.io returns 404).

**Pull request merged**